documentation/modules/upgrading/con-upgrade-listeners.adoc

```diff
@@ -0,0 +1,63 @@
+// This module is included in the following assemblies:
+//
+// upgrading/assembly_upgrade-kafka-versions.adoc
+
+[id='con-upgrade-listeners-{context}']
+= Updating listener configuration
+
+Strimzi provides a `GenericKafkaListener` schema for the configuration of Kafka listeners in a `Kafka` resource.
+
+`GenericKafkaListener` replaces the `KafkaListeners` schema, which is deprecated.
+
+The `GenericKafkaListener` schema is backwards compatible.
+
+With the `GenericKafkaListener` schema, you can configure as many listeners as required,
+as long as their names and ports are unique.
+The listener configuration is defined as an array.
+
+For clients inside the Kubernetes cluster, you can create `plain` (without encryption) or `tls` _internal_ listeners.
+
+For clients outside the Kubernetes cluster, you create _external_ listeners and specify a connection mechanism,
+which can be `nodeport`, `loadbalancer`, `ingress` or `route`.
+
+The `KafkaListeners` schema uses sub-properties for `plain`, `tls` and `external` listeners, with fixed ports for each.
+As part of your upgrade, you can convert listeners configured using the `KafkaListeners` schema into the format of the `GenericKafkaListener` schema.
```

Comment:
_After the upgrade_ would be better I guess since it gives the reader a clear time window.

Reply:
Changed to: _After the Kafka upgrade_